wp-includes/widgets/class-wp-widget-media-gallery.php

```diff
@@ -89,10 +89,6 @@ public function get_instance_schema() {
 				'media_prop'            => '_orderbyRandom',
 				'should_preview_update' => false,
 			),
-			'attachments' => array(
-				'type'                  => 'string',
-				'default' => '',
-			),
```

Comment:
Shouldn't this be retained, but have an `array` type with a default of `array()`?

Reply:
I'm new to these media widgets, but I don't think we need attachments in the schema at all unless there's some way to keep particular schema from being turned into input fields which get saved to the widget data (which is what we're trying to avoid).

Reply:
The schema is used by the widget's `update` method to handle sanitization and validation of the instance properties: https://github.com/xwp/wp-core-media-widgets/blob/1089ca34acba917e303f4b8a72e72284d63135cb/wp-includes/widgets/class-wp-widget-media.php#L247-L272

Reply:
And actually, the property could be:

```php
'attachments' => array(
    'type' => 'array',
    'default' => array(),
    'items' => 'integer',
),
```

To make it an explicit array of integer values.

Reply:
Aside: the Travis build is failing because `attachments` is not in the schema: https://travis-ci.org/xwp/wp-core-media-widgets/jobs/272758064#L650

Reply:
Right. We're already saving `ids` as part of the schema, so unless I'm missing something, I think we really do want to remove attachments from the schema, because we're not saving any attachment data, instead, we're generating the attachment data in the preview only.

Reply:
Oh 🤦‍♂️ you're right! Sorry, I missed that.

So yes, you're right to remove `attachments`. So then I think it is the `ids` that can be changed to have an `array` type if we want:

```php
'ids' => array(
    'type' => 'array',
    'default' => array(),
    'items' => 'integer',
),
```

Reply:
Currently, it looks like schema doesn't support an array type, because we would need to have some way to transform an array of data to/from the hidden input field that is generated for the widget form.

Reply:
Good point. The widget field syncing is assuming scalar values for the model. Alas, someday we'll get proper models for ~widgets~blocks.